File: 2324/Lab00/Schueler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Lab0{
    public class Schule
    {
        private string vorname = "";
        public void setVorname(string vorname)
        {
            this.vorname = vorname;
        }
        public string getVorname()
        {
            return this.vorname;
        }
        // C# mit properties
        private string _nachname; //code convention field

        public Schule(string vorname = "ur mum", int alter = 25)
        {
            this.vorname = vorname;
            _nachname = "AAAAAAAAA";
            Alter = alter;
        }

        public string Nachname {
            get
            {  return this._nachname; }
             set
            {
                if(value.Length < 2) {
                    throw new ArgumentException("nachname zu kurz");
                }
                _nachname = value;
            } 
        }
        public int Alter { get; set; }

        public override string ToString()
        {
            return Nachname +" "+getVorname()+" "+Alter;
        }
    }
  
}

File: 22.23/POS22.23/UE-01/src/main/java/Schueler.java
public class Schueler {
    private String nachname;
    private String vorname;
    private int alter;

    public Schueler(){
        setAlter(15);
        setNachname("Hödel");
        setVorname("Monika");
    }
    public Schueler(int alter, String vornahme, String nachname){
        setVorname(vornahme);
        setNachname(nachname);
        setAlter(alter);
    }

    public int getAlter() {
        return alter;
    }

    public String getNachname() {
        return nachname;
    }

    public String getVorname() {
        return vorname;
    }

    public void setAlter(int alter) {
        if(alter>5){
            this.alter = alter;
        }else{
            this.alter=15;
        }
    }

    public void setNachname(String nachname) {
        this.nachname = nachname;
    }

    public void setVorname(String vorname) {
        this.vorname = vorname;
    }

    public String toString(){
        return "Vorname:"+vorname+" Nachname:"+nachname+" Alter:"+alter;
    }
}
